### PR TITLE
✨ Some refactorings of the edge-scheduler

### PR DESCRIPTION
**Pull request by waltforme**
## Summary
This PR introduces some refactorings of the edge scheduler.
- change (file, function, etc.) names that inherited from the TMC controller;
- forget about namespaces, since we decided that the edge scheduler doesn't evaluate the 'what' part of scheduling;
- forget about TMC placements;
- ensure a SinglePlacementSlice object for each EdgePlacement object.

### CODE REVIEW

**Comment by ezrasilvera** [comment]
I'm a little confused about the name changes. In the code we switched from placement to scheduler, however in the package we moved from placement to edgeplacement under scheduling ?

File: pkg/reconciler/scheduling/edgeplacement/controller.go [hunk 1]
```diff
@@ -183,68 +133,21 @@ func (c *controller) enqueuePlacement(obj interface{}) {
...
+	// 	return
+	// }
+	// clusterName, _, _, err := kcpcache.SplitMetaClusterNamespaceKey(key)
+	// if err != nil {
+	// 	runtime.HandleError(err)
+	// 	return
+	// }
+
+	// TODO: We may or may not need to enqueue edgeplacements because of changes from locations.
+	// That depends on our future design.
```
Comment (MikeSpreitzer) [inline pkg/reconciler/scheduling/edgeplacement/controller.go:150]: This process should react to Location changes as well as Placement changes.  Perhaps use two workqueues, one for each kind of object, or use one workqueue with object kind included in the object reference values put into the queue.  Or, as does the old code here, use just one queue that holds only references to placements and bluntly enqueue references to all that could matter.

File: pkg/reconciler/scheduling/edgeplacement/controller.go [hunk 2]
```diff
@@ -305,7 +208,7 @@ func (c *controller) process(ctx context.Context, key string) error {
 	obj, err := c.edgePlacementLister.Cluster(clusterName).Get(name)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			logger.Info("object deleted before handled")
+			logger.V(2).Info("object deleted before handled")
```
Comment (MikeSpreitzer) [inline pkg/reconciler/scheduling/edgeplacement/controller.go:211]: Is this logger already at an elevated V level or does this statement actually correspond with `-v=2` ?  I think that this particular message is too inconsequential to appear at `-v=2`; I would make it 3 or a higher number.

**Review by MikeSpreitzer — COMMENTED** [review]
Looks mostly good.  A couple of comments inline.

File: pkg/reconciler/scheduling/edgeplacement/controller.go (hunk 1 above)
Comment (waltforme) [reply pkg/reconciler/scheduling/edgeplacement/controller.go]: Changed the comment to documents this. Thanks for suggesting these approaches!

File: pkg/reconciler/scheduling/edgeplacement/controller.go (hunk 2 above)
Comment (waltforme) [reply pkg/reconciler/scheduling/edgeplacement/controller.go]: Just checked the logger, its `.level` field equals 0 before calling `V`. Changed to 3.

**Review by MikeSpreitzer — APPROVED** [review]
/lgtm

**Pull request merged**